src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs

```diff
@@ -135,7 +136,12 @@ private static PortableExecutableReference MakeAssemblyMetadata(ModuleMetadata m
                         // or if the module has not been loaded yet. The value will be null
                         // if the name was ambiguous.
                         ModuleMetadata module;
-                        if (modulesByName.TryGetValue(name, out module) && (module != null))
+                        if (!modulesByName.TryGetValue(name, out module))
+                        {
+                            // AssemblyFile names may contain file information (".dll", etc).
+                            modulesByName.TryGetValue(Path.GetFileNameWithoutExtension(name), out module);
```

Comment:
As discussed offline, however, we could restrict this to some well known extensions:
.dll
.exe
.netmodule
.winmd